client/src/contexts/AuthContext.jsx

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext, useState, useEffect } from "react";
+import { supabase } from "../utils/supabaseClient";
+
+const AuthContext = createContext();
+
+export const AuthProvider = ({ children }) => {
+    const [user, setUser] = useState(null);
+
+    useEffect(() => {
+        const getSession = async () => {
+            const { data: { user } } = await supabase.auth.getUser();
+            setUser(user);
+        };
+        getSession();
+
+        const { data: listener } = supabase.auth.onAuthStateChange((_event, session) => {
+            setUser(session?.user || null);
```

Comment:
`||` works, but `??` is better since `??` will always return the right if the left is null. Whereas `||` treats 0 as false value.
```suggestion
            setUser(session?.user ?? null);
```